src/server/pkg/deploy/images/images.go

```diff
@@ -101,18 +101,17 @@ func Import(opts *assets.AssetOpts, in io.Reader) error {
 				},
 				authConfig,
 			); err != nil {
-				loopErr = append(loopErr, fmt.Errorf("error pushing to %s: %v", registry, err))
+				loopErr = append(loopErr, errors.Wrapf(err, "error pushing to %s", registry))
 				continue
 			}
 			pushed = true
 			break
 		}
 		if !pushed {
-			errStr := ""
-			for _, err := range loopErr {
-				errStr += err.Error() + "\n"
+			if len(loopErr) > 0 {
+				return loopErr[0]
```

Comment:
We actually don't want to short circuit here, because what we're doing is using the different configs to try to push to a repository. So it's possible that the 3rd config happens to be the one that can actually authorize the push, but the first 2 never would have been able to. I'll implement first suggestion wrapping and mentioning how many there were.